lib/websockex.ex

```diff
@@ -1103,11 +1109,11 @@ defmodule WebSockex do
       {:ok, new_state} ->
         {:ok, new_state}
 
-      {:reconnect, new_state} ->
-        {:reconnect, state.conn, new_state}
+      {:backoff, timeout, new_conn, new_state} ->
+        {:backoff, timeout, new_conn, new_state}
 
-      {:reconnect, new_conn, new_state} ->
-        {:reconnect, new_conn, new_state}
+      {:backoff, timeout, new_state} ->
+        {:backoff, timeout, state.conn, new_state}
```

Comment:
TBH it doesn't necessarily even have to be called `:backoff`. I don't think that it would really be confusing to just have:
```
{:reconnect, timeout, state}
```
Would you prefer to keep the `reconnect` word rather than `backoff`?

What I don't want though is to keep the existing behaviour of reconnect which basically does a `while true: reconnect`.
IMO this is one of those footguns that shouldn't exist. It results in spamming the server with reconnects and I've found that if a library enables bad behaviour by default, then this behaviour will end up occurring in the wild and make some poor devs/SREs very unhappy.

Of course a user could opt into this bad behaviour by doing
```
{:reconnect, 0, state}
```
But at least that's an explicit choice and in an explicit choice, a user might be more inclined to specify a timeout of 1s or something.

If we want to keep the existing result with no timeout `{:reconnect, state} ` then IMO we should change the behaviour to add a default backoff of at least 1s or something. Of course this could also be considered a breaking change and I would personally prefer making the explicit breaking change of requiring an explicit timeout so that users opt into the change of behaviour by making the small refactor of  `{:reconnect, state} ` to `{:reconnect, timeout, state} `.

Reply:
I can buy that.

I would say keep `:reconnect` and default `{:reconnect, state}` to `{:reconnect, default, state}`. That way the breaking change is the handling of messages.

I also think that we may need to nest `receive` blocks to catch the reconnect message as a priority.